### PR TITLE
arch: rename 'SRAM' memory regions to 'RAM' where appropriate

### PR DESCRIPTION
**Pull request by tari**
The name of the default RAM region for the linker varies somewhat between architectures, generally either being `RAM` or `SRAM`. Since it's useful at a higher level to have predictable memory region names (especially for the `CODE_DATA_RELOCATION` feature), this renames `SRAM` regions to `RAM` for architectures where that makes sense.

`RAM` is chosen as the unified name because `SRAM` may not be technically correct for some systems. Some SoCs and architectures have more complex memory layouts so they remain unchanged: they do not currently support `CODE_DATA_RELOCATION` which provides the motivation.

This is a follow-up from #49262 to clean up some of the build configuration.

### CODE REVIEW

**Review by edersondisouza — DISMISSED** [review]

File: include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld [hunk 1]
```diff
@@ -21,10 +21,10 @@
 /* physical address of RAM */
 #ifdef CONFIG_XIP
   #define ROMABLE_REGION FLASH
-  #define RAMABLE_REGION SRAM
+  #define RAMABLE_REGION RAM
 #else
-  #define ROMABLE_REGION SRAM
-  #define RAMABLE_REGION SRAM
+  #define ROMABLE_REGION RAM
+  #define RAMABLE_REGION RAM
```
Comment (SgrrZhf) [inline include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld:27]: As they both have `#define RAMABLE_REGION RAM`, could we put `#define RAMABLE_REGION RAM` out of the #ifdef preprocessor?

**Review by carlocaione — DISMISSED** [review]
LGTM

File: include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld (hunk 1 above)
Comment (tari) [reply include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld]: Done.

**Review by carlocaione — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Review by edersondisouza — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by SgrrZhf — DISMISSED** [review]

**Review by tejlmand — CHANGES_REQUESTED** [review]
what about:
https://github.com/zephyrproject-rtos/zephyr/blob/760c66688b845a4c8bd6ad1e73081cff8166d3f2/soc/arm/nxp_imx/rt5xx/CMakeLists.txt#L25
https://github.com/zephyrproject-rtos/zephyr/blob/8c8367d478d82f0f9a9ed84d63a908f6e8277d95/samples/application_development/code_relocation_nocopy/CMakeLists.txt#L19

?

Also, building `samples/application_development/code_relocation_nocopy` warns:
```
[172/182] Linking C executable zephyr/zephyr_pre0.elf
/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd:zephyr/linker_zephyr_pre0.cmd:84: warning: memory region `SRAM' not declared

[176/182] Linking C executable zephyr/zephyr_pre1.elf
/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd:zephyr/linker_zephyr_pre1.cmd:84: warning: memory region `SRAM' not declared

[182/182] Linking C executable zephyr/zephyr.elf
/opt/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd:zephyr/linker.cmd:84: warning: memory region `SRAM' not declared
Memory region         Used Size  Region Size  %age Used
        EXTFLASH:          64 B         8 MB      0.00%
           FLASH:       32580 B         1 MB      3.11%
             RAM:        6568 B       448 KB      1.43%
        IDT_LIST:          0 GB         2 KB      0.00%
       *default*:        7340 B 18446744073709551615 B      0.00%
$
```
Note, failed in other toolchain.

**Comment by tari** [comment]
> what about:
>
> ...

Good catch; fixed those.

> Also, building `samples/application_development/code_relocation_nocopy` warns:

That's a symptom of referring to a region that doesn't exist, so it's fixed by the cmake change to that sample.

**Comment by tejlmand** [comment]
> That's a symptom of referring to a region that doesn't exist, so it's fixed by the cmake change to that sample.

I know, it was just an extra piece of info for you :wink:

**Review by tejlmand — DISMISSED** [review]
lgtm

**Review by SgrrZhf — DISMISSED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
Should this be considered a bugfix? (would queue for v3.3.otherwise since we are in feature freeze)

**Comment by tari** [comment]
I wouldn't call this a bugfix, since it doesn't affect normal usage. It's even a sort of breaking change, since users referring to `SRAM` in a `zephyr_code_relocate` will need to change to `RAM` on changed platforms.

**Review by tejlmand — APPROVED** [review]

**Review by SgrrZhf — APPROVED** [review]

**Pull request merged**